Gems/Atom/RHI/Code/Include/Atom/RHI/FrameAttachment.h

```diff
@@ -78,9 +88,9 @@ namespace AZ::RHI
         AttachmentLifetimeType m_lifetimeType;
         HardwareQueueClassMask m_usedQueueMask = HardwareQueueClassMask::None;
         HardwareQueueClassMask m_supportedQueueMask = HardwareQueueClassMask::None;
-        ScopeAttachment* m_firstScopeAttachment = nullptr;
-        ScopeAttachment* m_lastScopeAttachment = nullptr;
-        Scope* m_firstScope = nullptr;
-        Scope* m_lastScope = nullptr;
+        AZStd::unordered_map<int, ScopeAttachment*> m_firstScopeAttachments;
+        AZStd::unordered_map<int, ScopeAttachment*> m_lastScopeAttachments;
```

Comment:
Would it be possible to move these pointers into one structure and just have one single map, reducing the memory requirements of the `FrameAttachment`?

Reply:
Done.